frame/support/procedural/src/pallet/mod.rs

```diff
@@ -31,21 +31,34 @@ mod parse;
 pub use parse::Def;
 use syn::spanned::Spanned;
 
+mod keyword {
+	syn::custom_keyword!(dev_mode);
+}
+
 pub fn pallet(
 	attr: proc_macro::TokenStream,
 	item: proc_macro::TokenStream,
 ) -> proc_macro::TokenStream {
+	let mut dev_mode = false;
 	if !attr.is_empty() {
-		let msg =
-			"Invalid pallet macro call: expected no attributes, e.g. macro call must be just \
-			`#[frame_support::pallet]` or `#[pallet]`";
-		let span = proc_macro2::TokenStream::from(attr).span();
-		return syn::Error::new(span, msg).to_compile_error().into()
+		if let Ok(_) = syn::parse::<keyword::dev_mode>(attr.clone()) {
+			dev_mode = true;
+		} else {
+			let msg = "Invalid pallet macro call: unexpected attribute. Macro call must be \
+				bare, such as `#[frame_support::pallet]` or `#[pallet]`, or must specify the \
+				`dev_mode` attribute, such as `#[frame_support::pallet(dev_mode)]` or \
+				#[pallet(dev_mode)]. No other attributes are supported at this time.";
+			let span = proc_macro2::TokenStream::from(attr).span();
+			return syn::Error::new(span, msg).to_compile_error().into()
+		}
 	}
 
 	let item = syn::parse_macro_input!(item as syn::ItemMod);
-	match parse::Def::try_from(item) {
-		Ok(def) => expand::expand(def).into(),
+	match parse::Def::try_from(item, dev_mode) {
+		Ok(mut def) => {
+			def.dev_mode = dev_mode;
```

Comment:
Why is this assignment required? The call to `Def::try_from` should have already assigned `dev_mode` to this field...

Reply:
Good catch. Originally it was set here forgot to remove